From 8cc409de0e1a2ab34b1400104b79447910e909b9 Mon Sep 17 00:00:00 2001
From: b18293 <b18293@freescale.com>
Date: Wed, 30 Jun 2010 15:27:08 +0800
Subject: [PATCH 01/58] create a git to put our demo java app

---
 readme | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 readme

diff --git a/readme b/readme
new file mode 100644
index 0000000..f2002ba
--- /dev/null
+++ b/readme
@@ -0,0 +1 @@
+It  is not only for E-Ink app, but also some other app we have done for demo usage. 
-- 
1.8.0

