From 2b3ba1fe13533463029cb428b7cf96d4768c9f12 Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Wed, 30 Jul 2014 18:47:10 +0300
Subject: [PATCH] Remove unneeded debug log message

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 wifi/wifi.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/wifi/wifi.c b/wifi/wifi.c
index 7165012..a9f74a3 100644
--- a/wifi/wifi.c
+++ b/wifi/wifi.c
@@ -659,7 +659,6 @@ int wifi_send_command(const char *cmd, char *reply, size_t *reply_len)
     if (!strncmp(cmd,NAMECHECK, strlen(NAMECHECK))) {
 		cmd_mod = (char *)&cmd[strlen(NAMECHECK)];
                 len -= strlen(NAMECHECK);
-        ALOGD("old command '%s', new command '%s'\n", cmd,cmd_mod);
 
     }
     ret = wpa_ctrl_request(ctrl_conn, cmd_mod, len, reply, reply_len, NULL);
-- 
1.7.9.5

