From e670bc65278c53a3ead9fc3867d11d3d3518ed76 Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Tue, 2 Aug 2011 20:28:40 +0800
Subject: [PATCH 05/10] ENGR00154182 update copyright

update copyright

Signed-off-by: b02247 <b02247@freescale.com>
---
 android/ip-down-ppp0.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/android/ip-down-ppp0.c b/android/ip-down-ppp0.c
index 8c6e624..1e54c22 100644
--- a/android/ip-down-ppp0.c
+++ b/android/ip-down-ppp0.c
@@ -13,6 +13,8 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+/* Copyright (C) 2010-2011 Freescale Semiconductor,Inc. */
+
 #include <unistd.h>
 #include <stdio.h>
 #include <stdlib.h>
-- 
1.8.0

