From 35d7f34e87bc41f00390fdc42b5875837066d178 Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Sun, 13 Jul 2014 11:13:45 +0300
Subject: [PATCH] Enlarge system partition to 700MB

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 cuboxi/BoardConfig.mk |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cuboxi/BoardConfig.mk b/cuboxi/BoardConfig.mk
index 5e0ca37..5dfa3f6 100755
--- a/cuboxi/BoardConfig.mk
+++ b/cuboxi/BoardConfig.mk
@@ -56,7 +56,7 @@ TARGET_USERIMAGES_SPARSE_EXT_DISABLED := true
 
 # uncomment below lins if use NAND
 #TARGET_USERIMAGES_USE_UBIFS = true
-BOARD_SYSTEMIMAGE_PARTITION_SIZE := 524288000
+BOARD_SYSTEMIMAGE_PARTITION_SIZE := 716800000
 
 ifeq ($(TARGET_USERIMAGES_USE_UBIFS),true)
 UBI_ROOT_INI := device/fsl/cuboxi/ubi/ubinize.ini
-- 
1.7.9.5

