From 8583eb5238b7bd26071077eaa257c978a51376fa Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Sun, 11 May 2014 22:26:57 +0300
Subject: [PATCH] CuBox-i and HummingBoard u-boot defconfigs

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 cuboxi/BoardConfig.mk |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cuboxi/BoardConfig.mk b/cuboxi/BoardConfig.mk
index 2e2c1a9..e198213 100755
--- a/cuboxi/BoardConfig.mk
+++ b/cuboxi/BoardConfig.mk
@@ -94,6 +94,6 @@ BOARD_HAVE_USB_CAMERA := true
 NUM_FRAMEBUFFER_SURFACE_BUFFERS := 3
 
 
-#TARGET_BOOTLOADER_CONFIG := 6q:mx6q_sabresd_android_config 6dl:mx6dl_sabresd_android_config
+TARGET_BOOTLOADER_CONFIG := mx6_cubox-i_config
 TARGET_KERNEL_DEFCONF := imx_v7_cubox-i_hummingboard_android_defconfig
 TARGET_KERNEL_DTB := imx6q-cubox-i.dtb imx6dl-cubox-i.dtb imx6dl-hummingboard.dtb
-- 
1.7.9.5

