From 995c14b7439fa3735c6d4bb80a53f9064e39fe6b Mon Sep 17 00:00:00 2001
From: Lin Fuzhen <fuzhen.lin@freescale.com>
Date: Fri, 15 Mar 2013 17:32:03 +0800
Subject: [PATCH 6/7] ENGR00254281: add ubifs support default

build libubi default

Signed-off-by: Lin Fuzhen <fuzhen.lin@freescale.com>
---
 Android.mk | 2 --
 1 file changed, 2 deletions(-)

diff --git a/Android.mk b/Android.mk
index 407f033..6571161 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,3 +1 @@
-ifeq ($(strip $(TARGET_USERIMAGES_USE_UBIFS)),true)
 include $(all-subdir-makefiles)
-endif
-- 
1.8.0

