From b2dfd7c9625bf85d0e67dc3dd7d73f53ba5627d5 Mon Sep 17 00:00:00 2001
From: Alan Tull <r80115@freescale.com>
Date: Tue, 18 Mar 2008 11:36:32 -0500
Subject: [PATCH 013/435] ENGR00064089 add wdog module back in

---
 module_test/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/module_test/Makefile b/module_test/Makefile
index 1c58990..4b026b4 100644
--- a/module_test/Makefile
+++ b/module_test/Makefile
@@ -4,7 +4,7 @@ ifneq ($(KERNELRELEASE),)
 
 # When adding modules, either add to the 'module_list' for all platfoms or add
 # below to the 'module_list +=...' list for each platform that needs it.
-module_list-y := mxc_rtic_test.o
+module_list-y := mxc_rtic_test.o mxc_wdog_tm.o
 
 module_list-$(CONFIG_MXC_PMIC_MC13783) += mxc_pm_test.o mxc_pmic_power_testmod.o
 module_list-$(CONFIG_MXC_IPC) += ipctestmod.o
-- 
1.8.0

