From a9ce2f871546fe8b77dfe41f127210c884ac27b2 Mon Sep 17 00:00:00 2001
From: Alan Tull <r80115@freescale.com>
Date: Mon, 7 Apr 2008 09:08:14 -0500
Subject: [PATCH 019/435] ENGR00030479 remove misc-testdatabase.txt

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index b4d40e9..66b5430 100644
--- a/Makefile
+++ b/Makefile
@@ -56,7 +56,7 @@ install_actual:
 	mkdir -p $(DESTDIR)
 	-rm -rf $(DESTDIR)/*
 	cp -rf $(OBJDIR)/* $(DESTDIR)
-	cp autorun.sh test-utils.sh autorun-suite.txt misc-testdatabase.txt $(DESTDIR)
+	cp autorun.sh test-utils.sh autorun-suite.txt $(DESTDIR)
 
 clean :
 	$(MAKE) -C $(TOPDIR)/test $@
-- 
1.8.0

