From 37a29fe692f5ba20fc302b4e33f7400ea48cc4fd Mon Sep 17 00:00:00 2001
From: JasonHua <Jason.Hua@freescale.com>
Date: Fri, 8 Jan 2010 14:00:35 +0800
Subject: [PATCH 014/498] Fix preview issue for imx51-3stack board.

Signed-off-by: Jason Hua <r65076@freescale.com>
---
 libcamera/CameraHal.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/libcamera/CameraHal.cpp b/libcamera/CameraHal.cpp
index d842037..e865f88 100644
--- a/libcamera/CameraHal.cpp
+++ b/libcamera/CameraHal.cpp
@@ -36,8 +36,15 @@
 #define MIN_HEIGHT          	144
 #define DISPLAY_WIDTH		480	//lcd width
 #define DISPLAY_HEIGHT		640	//lcd height
+
+#ifdef IMX51_3STACK
+#define PREVIEW_WIDTH	    	535	//default preview width
+#define PREVIEW_HEIGHT	    	400	//default preview height
+#else
 #define PREVIEW_WIDTH	    	635	//default preview width
 #define PREVIEW_HEIGHT	    	480	//default preview height
+#endif
+
 #define PREVIEW_FORMAT	    	"yuv422i"
 #define PREVIEW_FRAMERATE   	30
 #define PICTURE_WIDTH   	640 	//default picture width
@@ -683,7 +690,7 @@ if(mIsTakingPic == 0){
     //mParameters.getPreviewSize(&g_preview_height, &g_preview_width);
     g_preview_height = PREVIEW_WIDTH;
     g_preview_width = PREVIEW_HEIGHT;
-    g_display_top = 15;
+    g_display_top = 30;
     g_display_left = (g_display_width - g_preview_width) / 2;
 #else
     //mParameters.getPreviewSize(&g_preview_width, &g_preview_height);
-- 
1.8.0

