From 5cdaa28280cc792fd69aeb93e410db79a6526cb4 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Fri, 15 Jan 2010 10:41:44 +0800
Subject: [PATCH 016/486] Change to pan display.

---
 libgralloc/framebuffer.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libgralloc/framebuffer.cpp b/libgralloc/framebuffer.cpp
index 88776c2..2b8f5af 100755
--- a/libgralloc/framebuffer.cpp
+++ b/libgralloc/framebuffer.cpp
@@ -109,8 +109,8 @@ static int fb_post(struct framebuffer_device_t* dev, buffer_handle_t buffer)
         const size_t offset = hnd->base - m->framebuffer->base;
         m->info.activate = FB_ACTIVATE_VBL;
         m->info.yoffset = offset / m->finfo.line_length;
-        if (ioctl(m->framebuffer->fd, /*FBIOPUT_VSCREENINFO*/FBIOPAN_DISPLAY, &m->info) == -1) {
-            LOGE("FBIOPUT_VSCREENINFO failed");
+        if (ioctl(m->framebuffer->fd, FBIOPAN_DISPLAY, &m->info) == -1) {
+            LOGE("FBIOPAN_DISPLAY failed");
             m->base.unlock(&m->base, buffer); 
             return -errno;
         }
-- 
1.8.0

