From b5b30877815762733b546d412aae3f1595b16edc Mon Sep 17 00:00:00 2001
From: Kevin Yang <kevin.yang@freescale.com>
Date: Wed, 30 Dec 2009 10:04:57 +0800
Subject: [PATCH 019/498] update makefile to exclude c2d header file

---
 libcopybit/Android.mk | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/libcopybit/Android.mk b/libcopybit/Android.mk
index 30d0475..4d5c15f 100755
--- a/libcopybit/Android.mk
+++ b/libcopybit/Android.mk
@@ -13,6 +13,7 @@
 # limitations under the License.
 
 ifeq ($(BOARD_SOC_TYPE),IMX51)
+ifeq ($(PREBUILT_FSL_IMX_GPU),false)
 
 LOCAL_PATH := $(call my-dir)
 
@@ -22,8 +23,8 @@ include $(CLEAR_VARS)
 LOCAL_PRELINK_MODULE := false
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_C_INCLUDES += hardware/imx51/libgralloc
-LOCAL_C_INCLUDES += external/fsl-linux-amd-gpu/driver/include/private/C2D
 LOCAL_C_INCLUDES += kernel_imx/include
+LOCAL_C_INCLUDES += external/fsl-linux-amd-gpu/driver/include/private/C2D
 LOCAL_SHARED_LIBRARIES := liblog libc2d_z160
 
 LOCAL_SRC_FILES := 	\
@@ -34,3 +35,4 @@ LOCAL_CFLAGS:= -DLOG_TAG=\"$(TARGET_BOARD_PLATFORM).copybit\" -D_LINUX
 include $(BUILD_SHARED_LIBRARY)
 
 endif
+endif
-- 
1.8.0

