From 7724310995f7f28accac4531f7886563059a34ff Mon Sep 17 00:00:00 2001
From: JasonHua <Jason.Hua@freescale.com>
Date: Tue, 9 Mar 2010 17:08:06 +0800
Subject: [PATCH 022/502] Add D1 support for camcorder.

Signed-off-by: Jason Hua <r65076@freescale.com>
---
 libcamera/CameraHal.cpp | 13 ++-----------
 1 file changed, 2 insertions(+), 11 deletions(-)

diff --git a/libcamera/CameraHal.cpp b/libcamera/CameraHal.cpp
index 43af627..665fab1 100644
--- a/libcamera/CameraHal.cpp
+++ b/libcamera/CameraHal.cpp
@@ -726,17 +726,8 @@ if(mIsTakingPic == 0){
     mParameters.getPreviewSize(&width, &height);
     LOGI("Original Preview Width=%d Height=%d", width, height);
 
-    mParameters.getPreviewSize(&width, &height);
-    if( height < 288 ) {
-	g_recording_level = 0;
-	g_recording_width = RECORDING_WIDTH_LOW;
-	g_recording_height = RECORDING_HEIGHT_LOW;
-    } else {
-	g_recording_level = 1;
-	g_recording_width = RECORDING_WIDTH_NORMAL;
-	g_recording_height = RECORDING_HEIGHT_NORMAL;
-    }
-    LOGI("Recording width = %d, Recording height = %d", g_recording_width, g_recording_height);
+    g_recording_width = width;
+    g_recording_height = height;
 
 #ifdef IMX51_3STACK
     //mParameters.getPreviewSize(&g_preview_height, &g_preview_width);
-- 
1.8.0

