From d3c4ef7b313541ab60a84537adc6f890e4a86823 Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Sat, 13 Mar 2010 07:03:29 +0800
Subject: [PATCH 024/486] bug fix for copybit rotate 270

---
 libcopybit/copybit.cpp | 28 +++++++++++-----------------
 1 file changed, 11 insertions(+), 17 deletions(-)

diff --git a/libcopybit/copybit.cpp b/libcopybit/copybit.cpp
index 2167c7e..4a6c991 100755
--- a/libcopybit/copybit.cpp
+++ b/libcopybit/copybit.cpp
@@ -304,27 +304,13 @@ static void set_rects(struct copybit_context_t *dev,
     dstRect->height  = clip.b - clip.t;
 
     uint32_t W, H;
-    if (dev->mRotate == 90) {
+    if ((dev->mRotate == 90) || ((dev->mRotate == 270))) {
         srcRect->x = (clip.t - dst->t) + src->t;
         srcRect->y = (dst->r - clip.r) + src->l;
         srcRect->width = (clip.b - clip.t);
         srcRect->height = (clip.r - clip.l);
         W = dst->b - dst->t;
         H = dst->r - dst->l;
-    } else if (dev->mRotate == 180) {
-        srcRect->x = (clip.r - dst->r) + src->r;
-        srcRect->y = (dst->b - clip.b) + src->b;
-        srcRect->width = (clip.r - clip.l);
-        srcRect->height = (clip.b - clip.t);
-        W = dst->r - dst->l;
-        H = dst->b - dst->t;
-    } else if (dev->mRotate == 270) {
-        srcRect->x = (dst->b - clip.b) + src->b;
-        srcRect->y = (clip.l - dst->l) + src->r;
-        srcRect->width = (clip.b - clip.t);
-        srcRect->height = (clip.r - clip.l);
-        W = dst->b - dst->t;
-        H = dst->r - dst->l;    
     } else {
         srcRect->x  = (clip.l - dst->l) + src->l;
         srcRect->y  = (clip.t - dst->t) + src->t;
@@ -463,7 +449,8 @@ static int close_copybit(struct hw_device_t *dev)
         int fd;
         copybit_instance_t *ptr;
         C2D_STATUS c2dstatus;
-        c2dstatus = c2dDestroyContext(ctx->c2dctx);
+        if (ctx->c2dctx != NULL)
+        	c2dstatus = c2dDestroyContext(ctx->c2dctx);
         free(ctx);
 
         fd = open("/sqlite_stmt_journals/copybit.shm", O_RDWR, 0666);
@@ -539,9 +526,16 @@ static int open_copybit(const struct hw_module_t* module, const char* name,
     }
 
     if (status == 0) {
-        *device = &ctx->device.common;
         C2D_STATUS c2dstatus;        
         c2dstatus = c2dCreateContext(&ctx->c2dctx);             
+        if (c2dstatus != C2D_STATUS_OK)
+        {
+                close_copybit(&ctx->device.common);
+        }
+        else
+        {
+                *device = &ctx->device.common;            
+        }             
     } else {
         close_copybit(&ctx->device.common);
     }
-- 
1.8.0

