From a78c6758d16df120773de4d71a5632d159d85fdb Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Tue, 16 Mar 2010 10:24:05 +0800
Subject: [PATCH 025/486] Add vsync after pan display

---
 libgralloc/Android.mk      | 1 +
 libgralloc/framebuffer.cpp | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/libgralloc/Android.mk b/libgralloc/Android.mk
index 92a3868..cb5b980 100755
--- a/libgralloc/Android.mk
+++ b/libgralloc/Android.mk
@@ -21,6 +21,7 @@ include $(CLEAR_VARS)
 LOCAL_PRELINK_MODULE := false
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_SHARED_LIBRARIES := liblog libcutils libGLESv1_CM
+LOCAL_C_INCLUDES += kernel_imx/include
 
 LOCAL_SRC_FILES := 	\
 	allocator.cpp 	\
diff --git a/libgralloc/framebuffer.cpp b/libgralloc/framebuffer.cpp
index 2b8f5af..eb0e629 100755
--- a/libgralloc/framebuffer.cpp
+++ b/libgralloc/framebuffer.cpp
@@ -36,6 +36,7 @@
 
 #if HAVE_ANDROID_OS
 #include <linux/fb.h>
+#include <linux/mxcfb.h>
 #endif
 #include <GLES/gl.h>
 
@@ -114,6 +115,10 @@ static int fb_post(struct framebuffer_device_t* dev, buffer_handle_t buffer)
             m->base.unlock(&m->base, buffer); 
             return -errno;
         }
+
+        unsigned int ret;
+        ioctl(m->framebuffer->fd, MXCFB_WAIT_FOR_VSYNC, &ret);
+
         m->currentBuffer = buffer;
         
     } else {
-- 
1.8.0

