From 4b36f96abf4f6273cd219b8c980b3479fe129a69 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Mon, 10 May 2010 14:47:54 +0800
Subject: [PATCH 038/502] Fix the build include path

---
 libcopybit/Android.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libcopybit/Android.mk b/libcopybit/Android.mk
index 7a0d355..c7366a6 100755
--- a/libcopybit/Android.mk
+++ b/libcopybit/Android.mk
@@ -22,7 +22,7 @@ LOCAL_PATH := $(call my-dir)
 include $(CLEAR_VARS)
 LOCAL_PRELINK_MODULE := false
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
-LOCAL_C_INCLUDES += hardware/imx51/libgralloc
+LOCAL_C_INCLUDES += hardware/mx5x/libgralloc
 LOCAL_C_INCLUDES += kernel_imx/include
 LOCAL_C_INCLUDES += external/fsl-linux-amd-gpu/driver/include/private/C2D
 LOCAL_SHARED_LIBRARIES := liblog libcutils libc2d_z160
-- 
1.8.0

