From cffabdc1ec41b32b01ce414b57bdf24dd542bea2 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <xinyu.chen@freescale.com>
Date: Thu, 22 Jul 2010 15:55:46 +0800
Subject: [PATCH 054/498] Update the Android.mk due to GPU git renaming.

---
 libcopybit/Android.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libcopybit/Android.mk b/libcopybit/Android.mk
index c0d9f20..664c97c 100755
--- a/libcopybit/Android.mk
+++ b/libcopybit/Android.mk
@@ -23,7 +23,7 @@ include $(CLEAR_VARS)
 LOCAL_PRELINK_MODULE := false
 LOCAL_MODULE_PATH := $(TARGET_OUT_SHARED_LIBRARIES)/hw
 LOCAL_C_INCLUDES += hardware/mx5x/libgralloc
-LOCAL_C_INCLUDES += external/fsl-linux-amd-gpu/driver/include/private/C2D
+LOCAL_C_INCLUDES += external/fsl_imx_gpu/driver/include/private/C2D
 LOCAL_SHARED_LIBRARIES := liblog libcutils libc2d_z430
 
 LOCAL_SRC_FILES := 	\
-- 
1.8.0

