From 2b8ab5fe8b298568cf0187d8ff27963ded6b5c73 Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Fri, 2 Jul 2010 15:40:46 +0800
Subject: [PATCH 060/502] update framebuffer.cpp

---
 libgralloc/framebuffer.cpp | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/libgralloc/framebuffer.cpp b/libgralloc/framebuffer.cpp
index 7b47d61..61fbd6d 100755
--- a/libgralloc/framebuffer.cpp
+++ b/libgralloc/framebuffer.cpp
@@ -138,17 +138,17 @@ static void update_to_display(int left, int top, int width, int height, int upda
 	else 
         LOGI("waveform_mode  wrong\n");
 	   
-    if((updatemode & EINK_WAIT_MODE_MASK) == 1)
+    if((updatemode & EINK_WAIT_MODE_MASK) == 0x100)
         wait_for_complete = false;
-    else if((updatemode & EINK_WAIT_MODE_MASK) == 2)    
+    else if((updatemode & EINK_WAIT_MODE_MASK) == 0x200)    
         wait_for_complete = true;
     else 
         LOGI("wait_for_complete  wrong\n");
         
         
-    if((updatemode & EINK_AUTO_MODE_MASK) == 0)
+    if((updatemode & EINK_AUTO_MODE_MASK) == 0x0000)
         auto_update_mode = AUTO_UPDATE_MODE_REGION_MODE;
-    else if((updatemode & EINK_AUTO_MODE_MASK) == 1)    
+    else if((updatemode & EINK_AUTO_MODE_MASK) == 0x1000)    
         auto_update_mode = AUTO_UPDATE_MODE_AUTOMATIC_MODE;
     else 
         LOGI("wait_for_complete  wrong\n");
@@ -203,7 +203,7 @@ static int fb_setSwapInterval(struct framebuffer_device_t* dev,
 }
 
 static int fb_setUpdateRect(struct framebuffer_device_t* dev,
-        int l, int t, int w, int h, int updatemode)
+        int l, int t, int w, int h,int updatemode)
 {
     if (((w|h) <= 0) || ((l|t)<0))
         return -EINVAL;
-- 
1.8.0

