From da04bf756e762ebe111e6c2be90875862061de65 Mon Sep 17 00:00:00 2001
From: Richard Liu <r66033@freescale.com>
Date: Wed, 18 Aug 2010 08:17:01 +0800
Subject: [PATCH 068/486] copybit 180deg and 270deg rotation bug fix

---
 libcopybit/copybit.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libcopybit/copybit.cpp b/libcopybit/copybit.cpp
index 9b54ae0..ea80bfa 100755
--- a/libcopybit/copybit.cpp
+++ b/libcopybit/copybit.cpp
@@ -344,8 +344,8 @@ static void set_rects(struct copybit_context_t *dev,
     MULDIV(&srcRect->y, &srcRect->height, src->b - src->t, H);
 
     if ((dev->mRotate == 180) || ((dev->mRotate == 270))) {
-        srcRect->y = img->h - (srcRect->y + srcRect->height);
-        srcRect->x = img->w - (srcRect->x + srcRect->width);
+        srcRect->y = src->b - src->t - (srcRect->y + srcRect->height);
+        srcRect->x = src->r - src->l - (srcRect->x + srcRect->width);
     }
 }
 
-- 
1.8.0

