From e057a125472158ab64cebc46c6e6da354173871e Mon Sep 17 00:00:00 2001
From: Sammy He <r62914@freescale.com>
Date: Tue, 31 Aug 2010 01:36:02 +0800
Subject: [PATCH 069/486] Not stop preview when recording is completed.

Signed-off-by: Sammy He <r62914@freescale.com>
---
 libcamera/CameraHal.cpp | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/libcamera/CameraHal.cpp b/libcamera/CameraHal.cpp
index 37a3db9..9b1bd7e 100644
--- a/libcamera/CameraHal.cpp
+++ b/libcamera/CameraHal.cpp
@@ -399,6 +399,7 @@ int CameraHal::cameraTakePicConfig()
     parm.parm.capture.timeperframe.numerator = 1;
     parm.parm.capture.timeperframe.denominator = 15;
 #ifndef UVC_CAMERA
+    /* This capturemode value is related to ov3640 driver */
     if (mPictureWidth > 640 || mPictureHeight > 480)
         parm.parm.capture.capturemode = 3;  /* QXGA mode */
     else
@@ -827,7 +828,7 @@ status_t CameraHal::startPreview()
     mPreviewHeap.clear();
     for (i = 0; i< 3; i++)
         mPreviewBuffers[i].clear();
-    mPreviewHeap = new MemoryHeapBase(mRecordFrameSize * 3);
+    mPreviewHeap = new MemoryHeapBase(mRecordFrameSize * VIDEO_OUTPUT_BUFFER_NUM);
     for (i = 0; i < 3; i++)
        mPreviewBuffers[i] = new MemoryBase(mPreviewHeap, mRecordFrameSize * i, mRecordFrameSize);
 
@@ -927,7 +928,6 @@ void CameraHal::cameraPreviewStop()
 //    Mutex::Autolock lock(mLock);
     mPreviewShowFrameThread.clear();
     mPreviewCaptureFrameThread.clear();
-
 }
 
 bool CameraHal::previewEnabled()
@@ -948,8 +948,10 @@ status_t CameraHal::startRecording()
 
     LOGD("Clear the old memory ");
     mVideoHeap.clear();
-    for(i = 0; i < VIDEO_OUTPUT_BUFFER_NUM; i++)
+    for(i = 0; i < VIDEO_OUTPUT_BUFFER_NUM; i++) {
         mVideoBuffers[i].clear();
+        mVideoBufferUsing[i] = 0;
+    }
     LOGD("Init the video Memory %d", mRecordFrameSize);
     mVideoHeap = new MemoryHeapBase(mRecordFrameSize * VIDEO_OUTPUT_BUFFER_NUM);
     for(i = 0; i < VIDEO_OUTPUT_BUFFER_NUM; i++) {
@@ -970,7 +972,6 @@ void CameraHal::stopRecording()
     if(mMsgEnabled & CAMERA_MSG_PREVIEW_FRAME) {
         LOGD("Preview is still in progress\n");
     }
-    cameraPreviewStop();
 }
 
 bool CameraHal::recordingEnabled()
@@ -1282,7 +1283,7 @@ status_t CameraHal::setParameters(const CameraParameters& params)
     LOGD("mRecordWidth %d, mRecordHeight %d\n", mRecordWidth, mRecordHeight);
 
     mParameters.getPictureSize(&mPictureWidth, &mPictureHeight);
-    LOGD("mPictureWidth %d, mPictureHeight %d\n", mRecordWidth, mRecordHeight);
+    LOGD("mPictureWidth %d, mPictureHeight %d\n", mPictureWidth, mPictureHeight);
 
     if (!mPictureWidth || !mPictureHeight) {
         /* This is a hack. MMS APP is not setting the resolution correctly. So hardcoding it. */
-- 
1.8.0

