From c8957078733d52952707bdb63791d79a2009ff71 Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Thu, 2 Sep 2010 13:52:47 +0800
Subject: [PATCH 072/498] change the remperature to 24 for better performance

---
 libgralloc/framebuffer.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libgralloc/framebuffer.cpp b/libgralloc/framebuffer.cpp
index 998ed74..eb84fb0 100755
--- a/libgralloc/framebuffer.cpp
+++ b/libgralloc/framebuffer.cpp
@@ -159,7 +159,7 @@ static void update_to_display(int left, int top, int width, int height, int upda
 		LOGI("set auto update mode failed.  Error = 0x%x", retval);
 	}    
     
-    upd_data.temp = 0x8; //the temperature is get from wince team
+    upd_data.temp = 24; //the temperature is get from linux team
             
 	upd_data.update_region.left = left;
 	upd_data.update_region.width = width;
-- 
1.8.0

