From 778a863b0e202d142430ba6e7298e1d26d566c59 Mon Sep 17 00:00:00 2001
From: b07117 <b07117@freescale.com>
Date: Mon, 25 Jul 2011 16:51:26 +0800
Subject: [PATCH 167/498] ENGR00000000, fixed copybit_open return value

Signed-off-by: xianzhong <b07117@freescale.com>
---
 libcopybit/copybit.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/libcopybit/copybit.cpp b/libcopybit/copybit.cpp
index d5a6dd5..1f2a630 100755
--- a/libcopybit/copybit.cpp
+++ b/libcopybit/copybit.cpp
@@ -501,7 +501,10 @@ static int open_copybit(const struct hw_module_t* module, const char* name,
     if (c2dstatus != C2D_STATUS_OK)
         close_copybit(&ctx->device.common);
     else
+    {
         *device = &ctx->device.common;
+        status = 0;
+    }
 
     return status;
 }
-- 
1.8.0

