From 600be580664eb9b952d416dab96ea0b2a7e232d3 Mon Sep 17 00:00:00 2001
From: Xinyu Chen <b03824@freescale.com>
Date: Tue, 6 Sep 2011 10:28:20 +0800
Subject: [PATCH 172/486] fix typo for libsensors directory

Signed-off-by: Xinyu Chen <xinyu.chen@freescale.com>
---
 Android.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Android.mk b/Android.mk
index 2c989ae..9f1d376 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,4 +1,4 @@
-common_imx_dirs := libsensor alsa libcamera
+common_imx_dirs := libsensors alsa libcamera
 mx5x_dirs := $(common_imx_dirs) libgps mx5x/libcopybit mx5x/libgralloc
 mx6_dirs := $(common_imx_dirs) mx6/libgralloc
 
-- 
1.8.0

