From f8fe7b383784acfc4734c6dec6ac0a3f14089358 Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Sat, 3 May 2014 20:09:24 +0300
Subject: [PATCH] LK 3.10.x build instructions

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 core/Makefile |    8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/core/Makefile b/core/Makefile
index 01ba8de..d2e2b4c 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -811,9 +811,13 @@ $(TARGET_KERNEL_CONFIGURE): kernel_imx/arch/arm/configs/$(TARGET_KERNEL_DEFCONF)
 	install -D $(KERNEL_CONFIGURE) $(TARGET_KERNEL_CONFIGURE)
 
 $(TARGET_PREBUILT_KERNEL): $(TARGET_KERNEL_CONFIGURE)
-	$(MAKE) -C kernel_imx -j$(HOST_PROCESSOR) uImage $(KERNEL_ENV)
-	install -D $(KERNEL_UIMAGE)  $(PRODUCT_OUT)/uImage
+	$(MAKE) -C kernel_imx -j$(HOST_PROCESSOR) zImage $(TARGET_KERNEL_DTB) $(KERNEL_ENV)
 	install -D $(KERNEL_ZIMAGE)  $(PRODUCT_OUT)/kernel
+	install -D $(KERNEL_ZIMAGE)  $(PRODUCT_OUT)/zImage
+#	$(hide) $(foreach cf, $(TARGET_KERNEL_DTB),				\
+#	@echo "install dtb: $(cf)"						\
+#	install kernel_imx/arch/arm/boot/dts/$(cf) $(PRODUCT_OUT)/$(cf)		\
+#	)
 
 else
 kernelimage:
-- 
1.7.9.5

