From 355f4f663094fa5eafc9f1859fb75754b4850a6c Mon Sep 17 00:00:00 2001
From: lijian <jian.li@freescale.com>
Date: Wed, 16 Jan 2013 09:32:27 +0800
Subject: [PATCH 09/68] ENGR00133115 Support Subtitle

Signed-off-by: lijian <jian.li@freescale.com>
---
 include/media/OMXPlayer.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/include/media/OMXPlayer.h b/include/media/OMXPlayer.h
index 3fb6f32..cb1b2c1 100644
--- a/include/media/OMXPlayer.h
+++ b/include/media/OMXPlayer.h
@@ -69,8 +69,6 @@ public:
     virtual status_t    setParameter(int key, const Parcel &request);
     virtual status_t    getParameter(int key, Parcel *reply);
 
-    void                sendEvent(int msg, int ext1=0, int ext2=0) { MediaPlayerBase::sendEvent(msg, ext1, ext2); }
-
     status_t            ProcessEvent(int eventID, void* Eventpayload);
     status_t            ProcessAsyncCmd();
     status_t            PropertyCheck();
@@ -90,6 +88,7 @@ private:
     int                 mSharedFd;
     bool                bLoop;
     char                *contentURI;
+    char                *subtitleURI;
     MSGTYPE             msg;
     int                 msgData;
     void                *sem;
-- 
1.8.0

