From eab736841f77133093dd5c322ee0a013196d5b8e Mon Sep 17 00:00:00 2001
From: Xu Mao <B34688@freescale.com>
Date: Mon, 25 Nov 2013 13:51:13 +0800
Subject: [PATCH 44/50] ENGR00133115 use invoke to implement setPlaySpeed on
 kitkat

Signed-off-by: Xu Mao <B34688@freescale.com>
---
 include/media/mediaplayer.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/media/mediaplayer.h b/include/media/mediaplayer.h
index 4c05fc3..dd80a7b 100644
--- a/include/media/mediaplayer.h
+++ b/include/media/mediaplayer.h
@@ -170,6 +170,7 @@ enum media_player_invoke_ids {
     INVOKE_ID_SELECT_TRACK = 4,
     INVOKE_ID_UNSELECT_TRACK = 5,
     INVOKE_ID_SET_VIDEO_SCALING_MODE = 6,
+    INVOKE_ID_SET_PLAY_SPEED = 7,
 };
 
 // Keep MEDIA_TRACK_TYPE_* in sync with MediaPlayer.java.
-- 
1.8.0

