From 5023963b0c6b11a7009c10b74cef777994f75469 Mon Sep 17 00:00:00 2001
From: Xu Mao <B34688@freescale.com>
Date: Thu, 2 Jan 2014 14:52:05 +0800
Subject: [PATCH 51/72] ENGR00133115 add m2ts into support list of omx player

Signed-off-by: Xu Mao <B34688@freescale.com>
---
 media/libmediaplayerservice/MediaPlayerFactory.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/media/libmediaplayerservice/MediaPlayerFactory.cpp b/media/libmediaplayerservice/MediaPlayerFactory.cpp
index aad1ad6..5b00dba 100644
--- a/media/libmediaplayerservice/MediaPlayerFactory.cpp
+++ b/media/libmediaplayerservice/MediaPlayerFactory.cpp
@@ -338,6 +338,7 @@ class OMXPlayerFactory : public MediaPlayerFactory::IFactory {
                 ".mpg",
                 ".vob",
                 ".ts", 
+                ".m2ts", 
                 ".f4v",
                 ".mp3",
                 ".aac",
-- 
1.8.0

