From e61af3eb7eb682d38b8d65ddd9f63c0f258bfc37 Mon Sep 17 00:00:00 2001
From: Xu Mao <B34688@freescale.com>
Date: Fri, 7 Mar 2014 16:10:09 +0800
Subject: [PATCH 64/72] ENGR00133115 support mpeg2 decoding in stagefright

Signed-off-by: Xu Mao <B34688@freescale.com>
---
 media/libstagefright/OMXCodec.cpp | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/media/libstagefright/OMXCodec.cpp b/media/libstagefright/OMXCodec.cpp
index e0ce4e7..55fe541 100644
--- a/media/libstagefright/OMXCodec.cpp
+++ b/media/libstagefright/OMXCodec.cpp
@@ -1383,6 +1383,8 @@ void OMXCodec::setComponentRole(
             "video_decoder.avc", "video_encoder.avc" },
         { MEDIA_MIMETYPE_VIDEO_MPEG4,
             "video_decoder.mpeg4", "video_encoder.mpeg4" },
+        { MEDIA_MIMETYPE_VIDEO_MPEG2,
+            "video_decoder.mpeg2", "video_encoder.mpeg2" },
         { MEDIA_MIMETYPE_VIDEO_H263,
             "video_decoder.h263", "video_encoder.h263" },
         { MEDIA_MIMETYPE_VIDEO_VP8,
@@ -2300,9 +2302,9 @@ void OMXCodec::on_message(const omx_message &msg) {
                     mTargetTimeUs = -1;
                 }
 
-		mFilledBuffers.push_back(i);
-		mBufferFilled.signal();
-		if (mIsEncoder) {
+               mFilledBuffers.push_back(i);
+               mBufferFilled.signal();
+               if (mIsEncoder) {
                     sched_yield();
                 }
             }
@@ -4485,6 +4487,9 @@ void OMXCodec::initOutputFormat(const sp<MetaData> &inputFormat) {
             } else if (video_def->eCompressionFormat == OMX_VIDEO_CodingMPEG4) {
                 mOutputFormat->setCString(
                         kKeyMIMEType, MEDIA_MIMETYPE_VIDEO_MPEG4);
+            } else if (video_def->eCompressionFormat == OMX_VIDEO_CodingMPEG2) {
+                mOutputFormat->setCString(
+                        kKeyMIMEType, MEDIA_MIMETYPE_VIDEO_MPEG2);
             } else if (video_def->eCompressionFormat == OMX_VIDEO_CodingH263) {
                 mOutputFormat->setCString(
                         kKeyMIMEType, MEDIA_MIMETYPE_VIDEO_H263);
-- 
1.8.0

