From d6766058e903c06ea9b8bde6862e0b9b33548979 Mon Sep 17 00:00:00 2001
From: Xu Mao <B34688@freescale.com>
Date: Fri, 21 Mar 2014 13:20:35 +0800
Subject: [PATCH 66/68] ENGR00133115 change omxplayer calls into message

Signed-off-by: Xu Mao <B34688@freescale.com>
---
 include/media/OMXPlayer.h | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/include/media/OMXPlayer.h b/include/media/OMXPlayer.h
index abd1322..42e23ea 100644
--- a/include/media/OMXPlayer.h
+++ b/include/media/OMXPlayer.h
@@ -74,9 +74,18 @@ private:
     typedef enum {
         MSG_NONE,
         MSG_PREPAREASYNC,
+        MSG_START,
         MSG_SEEKTO,
+        MSG_PAUSE,
+        MSG_RESUME,
+        MSG_SETSURFACE,
     }MSGTYPE;
 
+    typedef struct {
+        MSGTYPE type;
+        int data;
+    }ASYNC_COMMAND;
+
     void                *player;
     status_t            mInit;
     sp<ANativeWindow>   mNativeWindow;
@@ -110,9 +119,15 @@ private:
     status_t            DoSeekTo(int msec);
     status_t            setVideoScalingMode(int32_t mode);
     status_t            getTrackInfo(Parcel *reply);
+    status_t            EnQueueCommand(ASYNC_COMMAND * pCmd);
+    status_t            DoSetVideoSurfaceTexture();
 
     bool                qdFlag;
     bool                bNetworkFail;
+    void                *cmdQueue;
+    bool                bPlaying;
+    void                *queueLock;
+    
 };
 
 typedef enum {
-- 
1.8.0

