From 435d70157ba94e97150207567d64b4f75bff5f0e Mon Sep 17 00:00:00 2001
From: Xu Mao <B34688@freescale.com>
Date: Thu, 28 Nov 2013 18:09:09 +0800
Subject: [PATCH] ENGR00133115 support mp4/3gp/f4v in http streaming

Signed-off-by: Xu Mao <B34688@freescale.com>
---
 .../android/webview/chromium/WebViewContentsClientAdapter.java | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/chromium/java/com/android/webview/chromium/WebViewContentsClientAdapter.java b/chromium/java/com/android/webview/chromium/WebViewContentsClientAdapter.java
index 566c99a..653c0b1 100644
--- a/chromium/java/com/android/webview/chromium/WebViewContentsClientAdapter.java
+++ b/chromium/java/com/android/webview/chromium/WebViewContentsClientAdapter.java
@@ -725,6 +725,16 @@ public class WebViewContentsClientAdapter extends AwContentsClient {
         if (mDownloadListener != null) {
             TraceEvent.begin();
             if (TRACE) Log.d(TAG, "onDownloadStart");
+
+            if(mimeType.equals("text/plain")){
+                if(url.endsWith(".mp4"))
+                    mimeType = "video/mp4";
+                else if(url.endsWith(".3gp"))
+                    mimeType = "video/3gp";
+                else if(url.endsWith(".f4v"))
+                    mimeType = "video/mp4";
+            }
+
             mDownloadListener.onDownloadStart(url,
                                               userAgent,
                                               contentDisposition,
-- 
1.8.0

