From d80a593b924249958d00e6fae9e69d0f66a3236a Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Sat, 3 May 2014 14:35:38 +0300
Subject: [PATCH] Kernel 3.10.30 signature

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 libc/kernel/common/linux/version.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libc/kernel/common/linux/version.h b/libc/kernel/common/linux/version.h
index 9c43d08..a0f7707 100644
--- a/libc/kernel/common/linux/version.h
+++ b/libc/kernel/common/linux/version.h
@@ -16,5 +16,5 @@
  ***
  ****************************************************************************
  ****************************************************************************/
-#define LINUX_VERSION_CODE 196643
+#define LINUX_VERSION_CODE 199198
 #define KERNEL_VERSION(a,b,c) (((a) << 16) + ((b) << 8) + (c))
-- 
1.7.9.5

