From 1c50577f7173878c4e043f0e25bd5b7e7202e5df Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Fri, 1 Jun 2012 09:54:53 +0800
Subject: [PATCH 03/12] ENGR00182565 No audio after playback music overnight

Failed in boundary

Signed-off-by: b02247 <b02247@freescale.com>
---
 pcm.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/pcm.c b/pcm.c
index 115f7d0..82e16fa 100644
--- a/pcm.c
+++ b/pcm.c
@@ -381,7 +381,7 @@ int pcm_get_htimestamp(struct pcm *pcm, unsigned int *avail,
 
     if (frames < 0)
         frames += pcm->boundary;
-    else if (frames > (int)pcm->boundary)
+    else if (frames >= (int)pcm->boundary)
         frames -= pcm->boundary;
 
     *avail = (unsigned int)frames;
@@ -802,7 +802,7 @@ static inline int pcm_mmap_playback_avail(struct pcm *pcm)
 
     if (avail < 0)
         avail += pcm->boundary;
-    else if (avail > (int)pcm->boundary)
+    else if (avail >= (int)pcm->boundary)
         avail -= pcm->boundary;
 
     return avail;
@@ -831,7 +831,7 @@ static void pcm_mmap_appl_forward(struct pcm *pcm, int frames)
     appl_ptr += frames;
 
     /* check for boundary wrap */
-    if (appl_ptr > pcm->boundary)
+    if (appl_ptr >= pcm->boundary)
          appl_ptr -= pcm->boundary;
     pcm->mmap_control->appl_ptr = appl_ptr;
 }
-- 
1.8.0

