From e33d25c575e007e73e1d579a2fa3e76ed72d6ed8 Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Tue, 8 Jul 2014 15:08:54 +0300
Subject: [PATCH] Install wpa_supplicant.conf on WiFi VER_0_8_X

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 wpa_supplicant/Android.mk |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wpa_supplicant/Android.mk b/wpa_supplicant/Android.mk
index 3c12665..54c6315 100644
--- a/wpa_supplicant/Android.mk
+++ b/wpa_supplicant/Android.mk
@@ -1599,7 +1599,7 @@ include $(BUILD_EXECUTABLE)
 #include $(BUILD_EXECUTABLE)
 #
 ########################
-ifeq ($(WPA_SUPPLICANT_VERSION),$(filter $(WPA_SUPPLICANT_VERSION),VER_0_8_UNITE))
+ifeq ($(WPA_SUPPLICANT_VERSION),$(filter $(WPA_SUPPLICANT_VERSION),VER_0_8_X))
 local_target_dir := $(TARGET_OUT)/etc/wifi
 
 include $(CLEAR_VARS)
-- 
1.7.9.5

